crates/ra_editor/src/typing.rs

```diff
@@ -136,6 +136,27 @@ pub fn on_eq_typed(file: &SourceFileNode, offset: TextUnit) -> Option<LocalEdit>
     })
 }
 
+pub fn on_dot_typed(file: &SourceFileNode, offset: TextUnit) -> Option<LocalEdit> {
+    let before_dot_offset = offset - TextUnit::of_char('.');
+
+    let _whitespace = find_leaf_at_offset(file.syntax(), before_dot_offset)
+        .left_biased()
+        .and_then(ast::Whitespace::cast)?;
+
```

Comment:
Here we should check that `.` is a part of chain call expression, and not just some random token, like this:

https://github.com/intellij-rust/intellij-rust/blob/ecc8def8829f86a0a6d20e3b0802ec463cf9174c/src/main/kotlin/org/rust/ide/typing/RsDotTypedHandler.kt#L36


I can't actually came up with a test where indenting dot would be *wrong* (and, indeed, IntelliJ has only positive tests), but it's safe to check this anyway.